boa_engine/src/builtins/string/mod.rs

```diff
@@ -2165,6 +2186,322 @@ impl String {
     }
 }
 
+#[cfg(feature = "annex-b")]
+impl String {
+    /// Helper function for creating html elements.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-createhtml
+    pub(crate) fn create_html(
+        string: &JsValue,
+        tag: &[u16],
+        attribute_and_value: Option<(&[u16], &JsValue)>,
+        context: &mut Context<'_>,
+    ) -> JsResult<JsValue> {
+        // 1. Let str be ? RequireObjectCoercible(string).
+
+        let str = string.require_object_coercible()?;
+
+        // 2. Let S be ? ToString(str).
+        let s = str.to_string(context)?;
+
+        // 3. Let p1 be the string-concatenation of "<" and tag.
+        let mut p1 = JsString::concat_array(&[utf16!("<"), tag]);
+        // 4. If attribute is not the empty String, then
+        if let Some((attribute, value)) = attribute_and_value {
+            // a. Let V be ? ToString(value).
+            let v = value.to_string(context)?;
+
+            // b. Let escapedV be the String value that is the same as V except that each occurrence
+            //    of the code unit 0x0022 (QUOTATION MARK) in V has been replaced with the six
+            //    code unit sequence "&quot;".
+            let mut escaped_v = Vec::with_capacity(v.len());
+            for c in v.as_slice().iter().copied() {
+                if c == 0x0022 {
+                    escaped_v.extend(utf16!("&quot;"));
+                    continue;
+                }
+                escaped_v.push(c);
+            }
+
+            // c. Set p1 to the string-concatenation of:
+            //    p1
+            //    the code unit 0x0020 (SPACE)
+            //    attribute
+            //    the code unit 0x003D (EQUALS SIGN)
+            //    the code unit 0x0022 (QUOTATION MARK)
+            //    escapedV
+            //    the code unit 0x0022 (QUOTATION MARK)
+            p1 = JsString::concat_array(&[
+                p1.as_slice(),
+                utf16!(" "),
+                attribute,
+                utf16!("=\""),
+                escaped_v.as_slice(),
+                utf16!("\""),
+            ]);
+        }
+
+        // 5. Let p2 be the string-concatenation of p1 and ">".
+        // 6. Let p3 be the string-concatenation of p2 and S.
+        // 7. Let p4 be the string-concatenation of p3, "</", tag, and ">".
+        let p4 = JsString::concat_array(&[
+            p1.as_slice(),
+            utf16!(">"),
+            s.as_slice(),
+            utf16!("</"),
+            tag,
+            utf16!(">"),
+        ]);
+        // 8. Return p4.
+        Ok(p4.into())
+    }
+    /// `String.prototype.anchor( name )`
```

Comment:
Let's add an empty line between all of the functions to make the code less cluttered.

```suggestion
    }

    /// `String.prototype.anchor( name )`
```